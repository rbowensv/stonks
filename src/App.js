import './App.scss';
import { ALPHA_VANTAGE_API, ALPHA_VANTAGE_ROUTE, APP_DESCRIPTION, APP_TITLE } from './app-constants';
import { StockSearch } from './StockSearch';
import { useStockSearch } from './useStockSearch';
import axios from 'axios';
import { StockView } from './StockView';

export const axiosInstance = axios.create({baseURL: ALPHA_VANTAGE_ROUTE, params: {apikey: ALPHA_VANTAGE_API}});

function App() {
  const {
    viewedStocks,
    searchSuggestions,
    getSearchSuggestion,
    pushStock,
    clearSearchSuggestions,
    removeStock
  } = useStockSearch();

  const stockSearchProps = {
      searchSuggestions,
      getSearchSuggestion,
      pushStock,
      clearSearchSuggestions,
      viewedStocks
  }

  return (
    <div className="App">
      <header className="App-header">
        {APP_TITLE}
        <div className="App-header--description">
          {APP_DESCRIPTION}
        </div>
        <StockSearch {...stockSearchProps}/>
      </header>
      <div className='stock-compare'>
        {viewedStocks && viewedStocks.map((stock) => {
          return <StockView key={stock.symbol} symbol={stock.symbol} name={stock.name} removeStock={removeStock}/>
        }
        )}
      </div>
    </div>
  );
}

export default App;
