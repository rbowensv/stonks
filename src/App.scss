.App{
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  justify-content: left;
  font-size: calc(10px + 2vmin);
  color: white;
  padding: 15px;
  height:100vh;
}
.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  justify-content: left;
  font-size: calc(10px + 2vmin);
  color: white;
  padding: 15px;

  &--description{
    font-size: calc(10px + 1vmin);
  }
}

@mixin a11y-visually-hidden {
  position: absolute !important;
  clip: rect(1px, 1px, 1px, 1px);
  overflow: hidden;
  height: 1px;
  width: 1px;
  word-wrap: normal;
}

.stock-search {
  &--suggestion{
    border-top: 0px !important;
    border-bottom:0px !important;
    padding-top:0px !important;
    padding-bottom:0px !important;

    &:hover{
      background-color:lightseagreen;
      color: white;
    }
  }

  &--label {
    @include a11y-visually-hidden();
  }

  &--suggestions{
    display: flex;
    flex-direction: column;
    max-width: 177px;

  }
}

.stock-compare{
  display:flex;
  align-items:stretch;
  align-content: stretch;
  justify-content: stretch;
  flex: 1 1 0px;

}

.stock-view{
  display:flex;
  flex-direction: column;
  flex: 1 1 0px;
  padding:5px;
  &--header {
    display:flex;
    justify-content: space-between;
  }
  font-size: 20px;
}

.stock-earnings{
  
  &--green{
    color:green;
  }

  &--red{
    color:red;
  }

  padding: 5px;

  &--row{
    display:flex;
  }

  &--section{
    display: flex;
    flex-direction: column;
    padding-left: 10px;


    > *{
      padding-right: 15px;
    }
  }

  &--meme{
    width:50px;
    height:50px;
  }
}

.stock-daily{
  padding: 5px;

  &--section{
    display:flex;
    > *{
      padding-right: 15px;

    }
  }
}

.stock-chart{
  max-width: 400px;
}