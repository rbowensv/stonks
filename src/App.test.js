import renderer from 'react-test-renderer'
import App from './App';

test('renders as expected', () => {
  const component = renderer.create(<App/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
