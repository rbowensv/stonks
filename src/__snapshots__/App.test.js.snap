// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    Stock Comparison
    <div
      className="App-header--description"
    >
      Enter up to 3 stocks to compare the current stock prices
    </div>
    <div
      className="stock-search"
    >
      <div>
        <label
          className="stock-search--label"
          htmlFor="stock-search"
        >
          Search stocks
        </label>
        <input
          className="stock-search--input"
          id="stock-search"
          onChange={[Function]}
          placeholder="Search..."
          type="text"
          value=""
        />
        <div
          className="stock-search--suggestions"
        />
      </div>
    </div>
  </header>
  <div
    className="stock-compare"
  />
</div>
`;
